OpenRA.Mods.Common/Widgets/ConfirmationDialogs.cs

```diff
@@ -101,18 +96,37 @@ public static class ConfirmationDialogs
 			Func<string, bool> inputValidator = null)
 		{
 			var panel = Ui.OpenWindow("TEXT_INPUT_PROMPT");
-			Func<bool> doValidate = null;
-			ButtonWidget acceptButton = null, cancelButton = null;
 
 			// Title
 			panel.Get<LabelWidget>("PROMPT_TITLE").GetText = () => title;
 
 			// Prompt
 			panel.Get<LabelWidget>("PROMPT_TEXT").GetText = () => prompt;
 
+			var acceptButton = panel.Get<ButtonWidget>("ACCEPT_BUTTON");
+			var cancelButton = panel.Get<ButtonWidget>("CANCEL_BUTTON");
+
 			// Text input
 			var input = panel.Get<TextFieldWidget>("INPUT_TEXT");
 			var isValid = false;
+
+			// Validation
+			Func<bool> doValidate = () =>
+			{
+				if (inputValidator == null)
+					return true;
```

Comment:
Don't we want to set `isValid` to `true` and `acceptButton.Disabled` to `false` if this is the case?

Reply:
I just moved this without trying to understand its behavior in detail, but: I think it shouldn't matter, since `Disabled` will never be set true in the first place if `inputValidator == null`.

Reply:
> I think it shouldn't matter, since `Disabled` will never be set true in the first place if `inputValidator == null`.

What about `isValid` which is initialised with `false`?